Blog on v4.2.0 (#22)
podium-lib/podium-lib.github.io: website/blog/2019-08-19-version-4.2.0.md

---
title: Version 4.2.0
author: Trygve Lie
authorURL: https://github.com/trygve-lie/
---

Version 4.2.0 of `@podium/layout` and `@podium/podlet` are now available.

### TypeDefinitions

> This release is shipped with TypeDefinitions for the public API.

Reviewer: ```suggestion
This release is shipped with TypeDefinitions for all public APIs.
```
sounds a bit better I think

### Assets

> Version 4.1.0 was a step on our path in paving ground for improving asset

Reviewer: ```suggestion
Version 4.1.0 was a step on the way to paving ground for improving the client side asset
```

> handling when building microfrontends with Podium.

Reviewer: ```suggestion
experience for developers building micro-frontends with Podium.
```

> This release contain a fair amount of changes to the `.js()` and `.css()`

Reviewer: ```suggestion
This release contains a number of changes to the `.js()` and `.css()`
```

> methods in both `@podium/layout` and `@podium/podlet`. The options on these

Reviewer: ```suggestion
methods in both `@podium/layout` and `@podium/podlet`. Possible options to these
```

> methods does now more or less correlates to the same attributes as on their

Reviewer: ```suggestion
methods now more or less correlate to the same attributes on the
```

equalent HTML elements.

> Example on how to flag an Javascript asset that it is an ES module and that it

Reviewer: ```suggestion
Example for how to flag a Javascript asset that it is an ES module so that it
```

can be loaded `async`:

```js
const podlet = new Podlet([ ... ]);

podlet.js({
    value: 'https://cdn.site.com/script.js',
    async: true,
    type: 'esm',
})
```

When rendered by the [`document template`](/docs/next/api/document.md) the above
will translate into the following HTML element:

```html
<script src="https://cdn.site.com/script.js" type="module" async></script>
```

Please see the following documentation for the options the different methods

> now can take:

Reviewer: ```suggestion 
can now take:
```

 * @podium/podlet [`.css()`](/docs/next/api/podlet#cssoptions-options) method
 * @podium/podlet [`.js()`](/docs/next/api/podlet#jsoptions-options) method
 * @podium/layout [`.css()`](/docs/next/api/layout#cssoptions-options) method
 * @podium/layout [`.js()`](/docs/next/api/layout#jsoptions-options) method

> For further information on how assets are handeled in general, please see the

Reviewer: ```suggestion
For further information on how assets are handled in general, please see the
```

[asset section](/docs/next/api/assets).

If you maintain a custom [`document template`](/docs/next/api/document), please
see [this section](/docs/next/api/document#assets) on how to render registered

> assets into HTML element respectively.

Reviewer: ```suggestion
assets into HTML elements appropriately.
```
